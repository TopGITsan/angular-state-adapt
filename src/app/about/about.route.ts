import { Route } from '@angular/router';

import {
  provideTranslocoLoadingTpl,
  provideTranslocoScope,
} from '@jsverse/transloco';
import { inlineLoader } from '@transloco/inline-loader.function';

export const ABOUT_ROUTE: Route = {
  path: 'about',
  loadComponent: () =>
    import('./about.component').then((m) => m.AboutComponent),
  providers: [
    provideTranslocoScope({
          scope: 'about',
          loader: inlineLoader(
            (lang: string, folder: string) => import(`./${folder}/${lang}.json`),
          ),
        }),
    provideTranslocoLoadingTpl(
      `<span id="default-loading-template">Loading template...</span>`,
    ),
  ],
};
