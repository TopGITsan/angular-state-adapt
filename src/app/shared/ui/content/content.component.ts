import { ChangeDetectionStrategy, Component } from '@angular/core';

@Component({
  selector: 'app-content',
  imports: [],
  template: ` <ng-content></ng-content> `,
  styles: `
    :host {
      @apply block w-full max-w-4xl m-auto p-4;
    }
  `,
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class ContentComponent {}
