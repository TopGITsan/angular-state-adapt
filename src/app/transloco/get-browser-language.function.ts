import { getBrowserLang } from '@jsverse/transloco';
import { isLang } from './is-lang.function';
import { fallbackLang } from './lang.state';
import { Lang } from './lang.types';

export function getBrowserLanguage(): Lang {
  const browserLanguage = getBrowserLang();
  const isSupported = isLang(browserLanguage);
  if (isSupported) {
    return browserLanguage as Lang;
  }
  return fallbackLang;
}
