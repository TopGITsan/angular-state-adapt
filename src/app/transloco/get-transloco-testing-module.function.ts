import {
  TranslocoTestingModule,
  TranslocoTestingOptions,
} from '@jsverse/transloco';

import aboutDE from '@about/i18n/de.json';
import aboutEN from '@about/i18n/en.json';
import aboutES from '@about/i18n/es.json';
import aboutFR from '@about/i18n/fr.json';
import aboutIT from '@about/i18n/it.json';
import aboutRO from '@about/i18n/ro.json';
import de from '../../assets/i18n/de.json';
import en from '../../assets/i18n/en.json';
import es from '../../assets/i18n/es.json';
import fr from '../../assets/i18n/fr.json';
import it from '../../assets/i18n/it.json';
import ro from '../../assets/i18n/ro.json';
import { langList } from './lang.constants';

export function getTranslocoTestingModule(options: TranslocoTestingOptions = {}) {
  const { langs, translocoConfig, ...rest } = options;
  return TranslocoTestingModule.forRoot({
    langs: {
      en,
      es,
      it,
      de,
      fr,
      ro,

      'about/es': aboutES,
      'about/en': aboutEN,
      'about/it': aboutIT,
      'about/fr': aboutFR,
      'about/de': aboutDE,
      'about/ro': aboutRO,

      ...langs,
    },
    translocoConfig: {
      availableLangs: langList,
      defaultLang: 'en',
      ...translocoConfig,
    },
    ...rest,
  });
}
