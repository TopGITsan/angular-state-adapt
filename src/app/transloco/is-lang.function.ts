import { langList } from "./lang.state";
import { Lang } from "./lang.types";

export function isLang(key: unknown): boolean {
  return !!key && typeof key === 'string' && langList.includes(key as Lang);
}
