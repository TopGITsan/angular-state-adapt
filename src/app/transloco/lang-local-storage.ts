import { languageLocalStorageKey } from './lang.constants';
import { Lang } from './lang.types';

export function getLanguageFromLocalStorage() {
  return localStorage.getItem(languageLocalStorageKey) ?? '';
}

export function setLanguageToLocalStorage(lang: Lang) {
  return localStorage.setItem(languageLocalStorageKey, lang);
}
