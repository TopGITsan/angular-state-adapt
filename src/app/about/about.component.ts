import { TitleCasePipe } from '@angular/common';
import { ChangeDetectionStrategy, Component } from '@angular/core';
import { MatButtonModule } from '@angular/material/button';
import {
  TranslocoDirective,
} from '@jsverse/transloco';
import { ExternalLinkDirective } from '@shared/directives';
import { ContentComponent } from '@shared/ui/content/content.component';

@Component({
  selector: 'app-about',
  imports: [
    ContentComponent,
    ExternalLinkDirective,
    MatButtonModule,
    TranslocoDirective,
    TitleCasePipe,
  ],
  template: `
    <ng-container *transloco="let t">
      <app-content>
        <h1>{{ t('about.about') | titlecase }}</h1>
        <p>
          <a mat-button href="https://angular.dev/">
            {{ title }}
          </a>
        </p>
        <p>
          <a mat-button href="https://state-adapt.github.io/angular">
            {{ subtitle }}
          </a>
        </p>
        <p>
          <a mat-button href="https://material.angular.io/">
            {{ components }}
          </a>
        </p>
        <p>
          <a mat-button href="https://tailwindcss.com/">
            {{ styles }}
          </a>
        </p>
      </app-content>
    </ng-container>
  `,
  styles: ``,
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class AboutComponent {
  readonly title = 'Build with angular';
  readonly subtitle = 'StateAdapt: Clean, Modern State Management';
  readonly styles =
    'Tailwind: A utility-first CSS framework packed with classes';
  readonly components =
    'Angular Material : Material Design components for Angular';
}
