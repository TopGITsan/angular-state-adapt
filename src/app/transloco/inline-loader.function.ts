import { InlineLoader, Translation } from '@jsverse/transloco';
import { langList } from './lang.state';

export const inlineLoader = (
  importer: (lang: string, folder: string) => Promise<Translation>,
  folder = 'i18n',
): InlineLoader =>
  langList.reduce((acc: InlineLoader, lang: string) => {
    acc[lang] = () => importer(lang, folder);
    return acc;
  }, {});
