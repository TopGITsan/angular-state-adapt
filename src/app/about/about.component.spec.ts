import { ComponentFixture, TestBed } from '@angular/core/testing';

import { TRANSLOCO_SCOPE } from '@jsverse/transloco';
import { ContentComponent } from '@shared/ui/content/content.component';
import { getTranslocoTestingModule } from '@internalization/get-transloco-testing-module.function';
import { AboutComponent } from './about.component';

describe('AboutComponent', () => {
  let component: AboutComponent;
  let fixture: ComponentFixture<AboutComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [AboutComponent, ContentComponent, getTranslocoTestingModule()],
      providers: [{ provide: TRANSLOCO_SCOPE, useValue: 'about' }],
    }).compileComponents();

    fixture = TestBed.createComponent(AboutComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
  it(`should have the 'Build with angular' title`, () => {
    const fixture = TestBed.createComponent(AboutComponent);
    const app = fixture.componentInstance;
    expect(app.title).toEqual('Build with angular');
  });

  it(`should have the 'StateAdapt: Clean, Modern State Management' subtitle`, () => {
    const fixture = TestBed.createComponent(AboutComponent);
    const app = fixture.componentInstance;
    expect(app.subtitle).toEqual('StateAdapt: Clean, Modern State Management');
  });

  it(`should have the 'Angular Material : Material Design components for Angular' subtitle`, () => {
    const fixture = TestBed.createComponent(AboutComponent);
    const app = fixture.componentInstance;
    expect(app.components).toEqual(
      'Angular Material : Material Design components for Angular',
    );
  });

  it(`should have the 'Tailwind: A utility-first CSS framework packed with classes' subtitle`, () => {
    const fixture = TestBed.createComponent(AboutComponent);
    const app = fixture.componentInstance;
    expect(app.styles).toEqual(
      'Tailwind: A utility-first CSS framework packed with classes',
    );
  });
});
