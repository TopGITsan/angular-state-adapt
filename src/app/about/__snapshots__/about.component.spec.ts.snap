// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutComponent should create 1`] = `
<app-about
  components={[Function String]}
  styles={[Function String]}
  subtitle={[Function String]}
  title={[Function String]}
>
  <app-content>
    <h1>
      About
    </h1>
    <p>
      <a
        class="mdc-button mat-mdc-button mat-unthemed mat-mdc-button-base"
        href="https://angular.dev/"
        mat-button=""
        mat-ripple-loader-class-name="mat-mdc-button-ripple"
        mat-ripple-loader-uninitialized=""
        target="_blank"
      >
        <span
          class="mat-mdc-button-persistent-ripple mdc-button__ripple"
        />
        <span
          class="mdc-button__label"
        >
           Build with angular 
        </span>
        <span
          class="mat-focus-indicator"
        />
        <span
          class="mat-mdc-button-touch-target"
        />
      </a>
    </p>
    <p>
      <a
        class="mdc-button mat-mdc-button mat-unthemed mat-mdc-button-base"
        href="https://state-adapt.github.io/angular"
        mat-button=""
        mat-ripple-loader-class-name="mat-mdc-button-ripple"
        mat-ripple-loader-uninitialized=""
        target="_blank"
      >
        <span
          class="mat-mdc-button-persistent-ripple mdc-button__ripple"
        />
        <span
          class="mdc-button__label"
        >
           StateAdapt: Clean, Modern State Management 
        </span>
        <span
          class="mat-focus-indicator"
        />
        <span
          class="mat-mdc-button-touch-target"
        />
      </a>
    </p>
    <p>
      <a
        class="mdc-button mat-mdc-button mat-unthemed mat-mdc-button-base"
        href="https://material.angular.io/"
        mat-button=""
        mat-ripple-loader-class-name="mat-mdc-button-ripple"
        mat-ripple-loader-uninitialized=""
        target="_blank"
      >
        <span
          class="mat-mdc-button-persistent-ripple mdc-button__ripple"
        />
        <span
          class="mdc-button__label"
        >
           Angular Material : Material Design components for Angular 
        </span>
        <span
          class="mat-focus-indicator"
        />
        <span
          class="mat-mdc-button-touch-target"
        />
      </a>
    </p>
    <p>
      <a
        class="mdc-button mat-mdc-button mat-unthemed mat-mdc-button-base"
        href="https://tailwindcss.com/"
        mat-button=""
        mat-ripple-loader-class-name="mat-mdc-button-ripple"
        mat-ripple-loader-uninitialized=""
        target="_blank"
      >
        <span
          class="mat-mdc-button-persistent-ripple mdc-button__ripple"
        />
        <span
          class="mdc-button__label"
        >
           Tailwind: A utility-first CSS framework packed with classes 
        </span>
        <span
          class="mat-focus-indicator"
        />
        <span
          class="mat-mdc-button-touch-target"
        />
      </a>
    </p>
  </app-content>
</app-about>
`;
